lib/json_schema/validator.rb

```diff
@@ -150,9 +151,14 @@ def validate_all_of(schema, data, errors, path)
 
     def validate_any_of(schema, data, errors, path)
       return true if schema.any_of.empty?
-      valid = schema.any_of.any? do |subschema|
-        validate_data(subschema, data, [], path)
+      subschemata_validity = schema.any_of.map do |subschema|
+        current_suberrors = []
+        current_valid = validate_data(subschema, data, current_suberrors, path)
+        @sub_errors << current_suberrors if current_suberrors.any?
```

Comment:
I think this probably shouldn't shovel onto `@sub_errors`, but that each `validate_...` method should take another param (`sub_errors`), as with `errors`...?